src/v/kafka/server/group_router.h

```diff
@@ -69,6 +69,33 @@ class group_router final {
           });
     }
 
+    template<typename Request, typename FwdFunc>
+    auto route_tx(Request&& r, FwdFunc func) {
+        // get response type from FwdFunc it has return future<response>.
+        using return_type = std::invoke_result_t<
+          FwdFunc,
+          decltype(std::declval<group_manager>()),
+          Request&&>;
+        using resp_type = typename return_type::value_type;
+
+        auto m = shard_for(r.group_id);
+        if (!m) {
+            resp_type reply;
+            reply.ec = cluster::tx_errc::not_coordinator;
```

Comment:
To support transactional processing I've added new intra-cluster RPCs: begin, prepare, commit & abort. Some of their erroneous state maybe described via kafka::error_code but some required new error codes and since we don't own kafka::error_code (it's part of the kafka protocol) I'm using internal error codes

Reply:
ah ok that makes sense. this sort of detail would be great to add to the commit message. really any sort of information like this should be added so that `git blame` can be used to trace a line of code back to some context / commentary.

Reply:
added